core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -650,6 +662,15 @@ private boolean hasTaskAsAncestor(Task<?> t, Task<?> potentialAncestor) {
         return hasTaskAsAncestor(t.getSubmittedByTask(), potentialAncestor);
     }
 
+    private Task<?> getRootTask(Task<?> t) {
+        Task<?> result = t;
+        while (t!=null) {
+            result = t;
+            t = t.getSubmittedByTask();
```

Comment:
Break the loop  if `getSubmittedByTask` returns same object reference as `t`.

Reply:
it is logically impossible and should be considered an illegal model for a task to claim to submit itself (or to have any recursive cycle here).  it would be ugly here if either of those happened (endless loop here) but it really shouldn't, easy to identify with `jstack`, and IMO isn't worth extra checks or tracking here to guard against an illegal model.

Reply:
I look at documentation
```
    /** task which submitted this task, if was submitted by a task */`
    public Task<?> getSubmittedByTask();
```
If task cannot submit itself then this should be fine of course.